android/java/org/chromium/chrome/browser/externalnav/BraveExternalNavigationHandler.java

```diff
@@ -29,7 +30,13 @@ public BraveExternalNavigationHandler(ExternalNavigationDelegate delegate) {
 
     @Override
     public OverrideUrlLoadingResult shouldOverrideUrlLoading(ExternalNavigationParams params) {
-        if (isWalletProviderOverride(params)) {
+        String originalUrl = params.getUrl().getSpec();
```

Comment:
Let's try passing these as `GURL`s as mentioned below.

Reply:
this should not require java at all if we're not having a url trigger an intent

Reply:
also just for general reference we should avoid duplicating even the kind of logic you have in here now. If we were going to do it this way, it should be something like `new_url 
= BraveRewardsNativeWorker.getInstance().maybeRedirectRewardsProtocol(url, referrerUrl)` where we'd override the url if `new_url` is not empty (optional return value would be better, but I don't think you can do that very easily in jni)